src/interface.ts

```diff
@@ -1,12 +1,20 @@
 type Context = Object;
+type Result = Object;
 type IntentName = String;
 type AppIdentifier = String;
+type AppInstanceIdentifier = String;
 
-enum OpenError {
+enum InvokationErrorType {
   AppNotFound = "AppNotFound",
   ErrorOnLaunch = "ErrorOnLaunch",
-  AppTimeout = "AppTimeout",
-  ResolverUnavailable = "ResolverUnavailable"
+  AppTimeout = "AppTimeout",  
+  HandlerInternalError = "HandlerInternalError",
```

Comment:
For "HandlerInternalError", what is the handler?  The intent handler on the App receiving the intent?  Why not call it "AppInternalError", and simplify the terminology?